### PR TITLE
asciidoc markup generator

### PR DESCRIPTION
**Pull request by man-at-home**
### PR checklist

- [x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [ x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- added a basic asciidoc markup documentation generator (oriented on html or cwiki) 

two advantages:
- a) printable output for openapi specs (pdf, more options via docbook) 
- b) generator allows to augment (include or link) the generated documentation with other outputs via mustache filters
- augmenting info would be: sample output by spring restdoc, hand written spec prose and in this case wiremock sample data generated by spring boot cloud contract

technical commitee: (core team as asciidoc markup has no team)

@wing328 @jimschubert @cbornet @ackintosh  @jmini @etherealjoy

### CODE REVIEW

**Comment by man-at-home** [comment]
need to fix several ci builds it seems (mostly style or windows specific stuff like \r) the next days..

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AsciidocDocumentationCodegen.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
+package org.openapitools.codegen.languages;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AsciidocDocumentationCodegen.java]: I'll add the copyright header in a separate PR.

File: samples/documentation/asciidoc/index.adoc [hunk 2]
```diff
@@ -0,0 +1,1659 @@
...
+
+
+== Endpoints
+
+
+[.Pet]
+=== Pet
+
+
+[.addPet]
```
Comment (wing328) [inline samples/documentation/asciidoc/index.adoc]: @man-at-home we may need to prefix it with the class name (e.g. Pet_addPet) as multiple classes can have the same method (e.g. both Pet and User have the same method `getId`: Pet_getId, User_getId).

**Comment by wing328** [comment]
FYI. I've added you as the template creator of asciidoc in 8f43373

File: samples/documentation/asciidoc/index.adoc (hunk 2 above)
Comment (man-at-home) [reply samples/documentation/asciidoc/index.adoc]: I will look into it. At the moment the generated doc does not use these tags, but the idee would be to allow links to specific calls (if the generated docs format is linkable like html)

**Comment by man-at-home** [comment]
thanks a lot.

**Comment by wing328** [comment]
@man-at-home do you have a Twitter account? We will tag you in a tweet to promote the new generator.

**Comment by man-at-home** [comment]
no twitter handle in use, sorry

**Comment by wing328** [comment]
That's ok. I've sent out the following:

https://twitter.com/oas_generator/status/1174992564576911360

https://news.ycombinator.com/item?id=21025034

https://www.reddit.com/r/opensource/comments/d6stws/generate_api_documentation_in_asciidoc_format/

Have a nice weekend!

**Comment by wing328** [comment]
@man-at-home thanks for the PR, which has been included in the v4.1.3 release: https://twitter.com/oas_generator/status/1180123829626003456

**Pull request merged**